python/cusignal/filtering/filtering.py

```diff
@@ -887,7 +899,11 @@ def hilbert(x, N=None, axis=-1):
         raise ValueError("N must be positive.")
 
     Xf = cp.fft.fft(x, N, axis=axis)
-    h = _hilbert_kernel(size=N)
+    
+    if Xf.dtype.char is "F":
```

Comment:
By using deduction, we no longer need this check.

We instead allocate an empty array for h, like
`h = cp.empty((N,), dtype=x.dtype)`

Reply:
I forgot to mention, now that we have `h` allocated, we only need the following

`_hilbert_kernel(h)`
and
`_hilbert2_kernel(h1, h2)`